pkg/hive/resources.go

```diff
@@ -98,17 +102,53 @@ func aroServiceKubeconfigSecret(namespace string, kubeConfig []byte) *corev1.Sec
 	}
 }
 
-func clusterServicePrincipalSecret(namespace string, secret []byte) *corev1.Secret {
-	return &corev1.Secret{
+func clusterAzureSecret(namespace string, oc *api.OpenShiftCluster, sub *api.SubscriptionDocument) (*corev1.Secret, error) {
+	secret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      clusterServicePrincipalSecretName,
 			Namespace: namespace,
 		},
-		Data: map[string][]byte{
-			"osServicePrincipal.json": secret,
-		},
+		Data: map[string][]byte{},
 		Type: corev1.SecretTypeOpaque,
 	}
+
+	// Add osServicePrincipal.json only when cluster is not managed identity
+	if !oc.UsesWorkloadIdentity() {
+		clusterSPBytes, err := clusterSPToBytes(sub, oc)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data["osServicePrincipal.json"] = clusterSPBytes
+	}
+
+	return secret, nil
+}
+
+func clusterManifestsSecret(namespace string, customManifests map[string]kruntime.Object) (*corev1.Secret, error) {
```

Comment:
Instead of changing the function name, changed the var name:- https://github.com/Azure/ARO-RP/pull/3841/files#diff-f83dd6c00cfc7b417ce96e6f7bfbe58ce118101ea90630f21d658abf3e96b129R60

`generateClusterManifestsSecret` name is similar to `https://github.com/Azure/ARO-RP/pull/3841/files#diff-71a1dd610bf07a9e4ece9f782af2bad11e6f2d0e03df442019088f8c47265497R281` where the actual generation is taking place and keeping it similar to similar type of secret creations in the file.